doc/source/api/core/partitioned.rst

```diff
@@ -0,0 +1,87 @@
+Partitioned Datasets
+====================
+
+The :class:`pyvista.PartitionedDataSet` class is a partitioned dataset that encapsulates
+a dataset consisting of partitions. ``PartitionedDataSet`` behaves mostly like a list.
+
+List-like Features
```

Comment:
I do not understand the list-like features. For me it's a tree-like structure. Then making it a list is reductive.

Reply:
Good point. The tree structure is a `PartitionedDataSetCollection`, which is a collection of `PartitionedDataSets`, and will be added in the next PR after this one.